scrapy/signalmanager.py

```diff
@@ -1,27 +1,71 @@
+from __future__ import absolute_import
 from scrapy.xlib.pydispatch import dispatcher
-from scrapy.utils import signal
+from scrapy.utils import signal as _signal
+
 
 class SignalManager(object):
 
     def __init__(self, sender=dispatcher.Anonymous):
         self.sender = sender
 
-    def connect(self, *a, **kw):
-        kw.setdefault('sender', self.sender)
-        return dispatcher.connect(*a, **kw)
+    def connect(self, receiver, signal, **kwargs):
+        """
+        Connect a receiver function to a signal.
+
+        The signal can be any object, although Scrapy comes with some
+        predefined signals that are documented in the :ref:`topics-signals`
+        section.
+
+        :param receiver: the function to be connected
+        :type receiver: callable
+
+        :param signal: the signal to connect to
+        :type signal: object
+        """
+        kwargs.setdefault('sender', self.sender)
+        return dispatcher.connect(receiver, signal, **kwargs)
+
+    def disconnect(self, receiver, signal, **kwargs):
+        """
+        Disconnect a receiver function from a signal. This has the
+        opposite effect of the :meth:`connect` method, and the arguments
+        are the same.
+        """
+        kwargs.setdefault('sender', self.sender)
+        return dispatcher.disconnect(receiver, signal, **kwargs)
+
+    def send_catch_log(self, signal, **kwargs):
+        """
+        Send a signal, catch exceptions and log them.
+
+        The keyword arguments are passed to the signal handlers (connected
+        through the :meth:`connect` method).
+        """
+        kwargs.setdefault('sender', self.sender)
+        return _signal.send_catch_log(signal, **kwargs)
+
+    def send_catch_log_deferred(self, signal, **kwargs):
+        """
+        Like :meth:`send_catch_log` but supports returning `deferred`_ from
+        signal handlers.
+
+        Returns a `deferred`_ that gets fired once all signal handlers
+        deferreds were fired. Send a signal, catch exceptions and log them.
 
-    def disconnect(self, *a, **kw):
-        kw.setdefault('sender', self.sender)
-        return dispatcher.disconnect(*a, **kw)
+        The keyword arguments are passed to the signal handlers (connected
+        through the :meth:`connect` method).
 
-    def send_catch_log(self, *a, **kw):
-        kw.setdefault('sender', self.sender)
-        return signal.send_catch_log(*a, **kw)
+        .. _deferreds: http://twistedmatrix.com/documents/current/core/howto/defer.html
```

Comment:
Shouldn't be `_deferred` (singular) here? Both mentions of this link are singular, but I think the first line of the docstring of `send_catch_log_deferred` should be changed too, it was "... but supports returning \`deferreds\`_ from ..." before.

Probably the links `_deferred` and `_deferreds`at the bottom of docs/topics/api.rst should be deleted.